server/server/src/main.rs

```diff
@@ -14,8 +16,11 @@ async fn main() -> std::io::Result<()> {
     let settings: Settings =
         configuration::get_configuration().expect("Failed to parse configuration settings");
 
+    let hardware_id = uuid().to_ascii_uppercase();
+    let app_data = AppData::write_to_file(hardware_id).expect("Failed to save hardware id to file");
```

Comment:
This looks like you are writing a new hardware id on every new start. Alternative could be something like:

```rust
let mut app_data = match AppData::load()?;
if app_data.is_empty() {
   app_data = AppData::write_to_file(uuid().to_ascii_uppercase()).expect("Failed to save hardware id to file")
}
```